doc/rst/usingchapel/compiling.rst

```diff
@@ -110,3 +110,92 @@ auto-complete the option and add a space, ready for the next option.
 
      % chpl --cac<tab>
      % chpl --cache-remote
+
+------------------------------
+Integrating Into Build Systems
+------------------------------
+
+Larger projects using Chapel may wish to setup a build system to manage
+compilation of their Chapel code. This section provides a brief overview of a
+few common build systems and how they can be used with Chapel.
+
+Mason
+^^^^^
+
+:ref:`Mason <readme-mason>` is the Chapel package manager and build system. See
+the :ref:`Mason Guide <index-mason-guide>` for a walkthrough of how to use
+Mason to manage your Chapel projects.
+
+Make
+^^^^
+
+If you wish to use Make to compile your Chapel code, you can specify a recipe
+that contains all of the Chapel files and flags in a single invocation. Chapel
+currently does not support separate compilation, so all Chapel files must be
+compiled together.
+
+For example, a simple ``Makefile`` that compiles a Chapel project might look
+like this. This can be extended with other flags or dependencies as needed:
+
+.. code-block:: Makefile
+
+   CHPL := chpl
+   DEBUG ?= 0
+   TARGET := myProgram
+
+   .PHONY: all
+   all: $(TARGET)
+
+   ifeq ($(DEBUG),1)
+   CHPL_FLAGS += -g
+   else
+   CHPL_FLAGS += --fast
+   endif
+
+   SOURCES := $(wildcard *.chpl)
+
+   $(TARGET): $(SOURCES)
+   	$(CHPL) $(CHPL_FLAGS) $< -o $@
+
+   .PHONY: clean
+   clean:
+   	rm -f $(TARGET)
+
+This can be used by running ``make`` in the directory containing the Makefile.
+Running ``make DEBUG=1`` will compile the program with debugging enabled and
+``make clean`` will remove the compiled program.
+
+CMake
```

Comment:
I'm not familiar enough with CMake to be able to evaluate this section, I'd grab someone on the dyno team.